hs_core/hydroshare/users.py

```diff
@@ -425,40 +423,21 @@ def get_resource_list(creator=None, group=None, user=None, owner=None, from_date
     if not include_obsolete:
         flt = flt.exclude(object_id__in=Relation.objects.filter(
             type='isReplacedBy').values('object_id'))
+    if full_text_search:
+        desc_ids = Description.objects.filter(abstract__icontains=full_text_search).values_list('object_id', flat=True)
+        title_ids = Title.objects.filter(value__icontains=full_text_search).values_list('object_id', flat=True)
+
+        # Full text search must match within the title or abstract
+        if desc_ids:
+            flt = flt.filter(object_id__in=desc_ids)
+        elif title_ids:
+            flt = flt.filter(object_id__in=title_ids)
+        else:
+            # No matches on title or abstract, so treat as no results of search
+            flt = flt.none()
```

Comment:
Not sure why we need `flt = flt.none()``` here. Wouldn't it suffice to leave ```flt``` as is, i.e., remove the ```else:``` condition?

Reply:
I simply moved this logic outside of the loop over the q array.  Looking this over, the description and title ids should be combined and then just applied to the filter.  I'll make the change.

Reply:
@hyi - see the latest commit